wizard/vote_on_onchain_governance.py

```diff
@@ -0,0 +1,208 @@
+import traceback
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import List
+
+from rich import print
+
+from wizard import errors, ux
+from wizard.accounts import load_accounts
+from wizard.configuration import CONFIGURATIONS
+from wizard.constants import DEFAULT_GAS_PRICE
+from wizard.entrypoint import MyEntrypoint
+from wizard.guardians import AuthApp
+from wizard.transactions import TransactionWrapper
+
+from multiversx_sdk.core.address import Address
+from multiversx_sdk.core.transactions_factory_config import TransactionsFactoryConfig
+from multiversx_sdk.governance.governance_transactions_factory import GovernanceTransactionsFactory
+from multiversx_sdk.governance.resources import VoteType
+from multiversx_sdk.network_providers.proxy_network_provider import ProxyNetworkProvider
+
+
+def _addr_from_bech32(b: str) -> Address:
+    if hasattr(Address, "new_from_bech32"):
+        return Address.new_from_bech32(b)
+    raise errors.KnownError("This SDK requires Address.new_from_bech32.")
+
+def _vote_enum(choice: str):
+    try:
+        return {"yes": VoteType.YES, "no": VoteType.NO, "abstain": VoteType.ABSTAIN}[choice]
+    except AttributeError:
+        return {"yes": getattr(VoteType, "yes"), "no": getattr(VoteType, "no"), "abstain": getattr(VoteType, "abstain")}[choice]
+
+
+def main(cli_args: list[str] = sys.argv[1:]):
+    try:
+        _do_main(cli_args)
+    except errors.KnownError as err:
+        ux.show_critical_error(traceback.format_exc())
+        ux.show_critical_error(err.get_pretty())
+        return 1
+
+def _do_main(cli_args: List[str]):
+    parser = ArgumentParser()
+    parser.add_argument("--network", choices=CONFIGURATIONS.keys(), required=True, help="network name")
+    parser.add_argument("--wallets", required=True, help="path to wallets configuration file")
+    parser.add_argument("--auth", required=True, help="auth registration file")
+    parser.add_argument("--gas-price", type=int, default=DEFAULT_GAS_PRICE, help="min gas price")
+    sub = parser.add_subparsers(dest="action", required=True)
+
+    vote_p = sub.add_parser("vote", help="cast votes (bulk) for a proposal")
+    vote_p.add_argument("--proposal", type=int, required=True, help="proposal nonce / id")
+    vote_p.add_argument("--choice", choices=["yes", "no", "abstain"], required=True, help="vote choice")
+
+    close_p = sub.add_parser("close", help="close a proposal")
+    close_p.add_argument("--proposal", type=int, required=True, help="proposal nonce / id")
+
+    propose_p = sub.add_parser("propose", help="create a new proposal")
+    propose_p.add_argument("--commit-hash", required=True, help="commit hash for proposal content")
+    propose_p.add_argument("--start-epoch", type=int, required=True, help="start vote epoch")
+    propose_p.add_argument("--end-epoch", type=int, required=True, help="end vote epoch")
+    propose_p.add_argument("--fee-egld", type=float, default=0.0, help="native EGLD amount to attach")
```

Comment:
this can be of type `int`.
the help could be modified a bit: `the fee to create a new proposal` or something similar